src/test/java/org/sagebionetworks/bridge/models/schedules2/adherence/study/StudyAdherenceReportGeneratorTest.java

```diff
@@ -248,80 +256,691 @@ public static Schedule2 createSchedule() {
+
+        List<EventStreamDay> week1Day1 = week1.getByDayEntries().get(1);
+        assertEquals(week1Day1.get(1).getTimeWindows().get(0).getState(), EXPIRED);
+        
+        List<EventStreamDay> week1Day2 = week1.getByDayEntries().get(2);
+        assertEquals(week1Day2.get(0).getTimeWindows().get(0).getState(), EXPIRED);
+        
+        StudyReportWeek week2 = report.getWeeks().get(1);
+        List<EventStreamDay> week2Day1 = week2.getByDayEntries().get(0);
+        assertEquals(week2Day1.get(0).getTimeWindows().get(0).getState(), EXPIRED);
+        
+        StudyReportWeek week3 = report.getWeeks().get(2);
+        List<EventStreamDay> week3Day1 = week3.getByDayEntries().get(0);
+        assertEquals(week3Day1.get(0).getTimeWindows().get(0).getState(), UNSTARTED);
+        
+        StudyReportWeek week4 = report.getWeeks().get(3);
+        List<EventStreamDay> week4Day1 = week4.getByDayEntries().get(0);
+        assertEquals(week4Day1.get(0).getTimeWindows().get(0).getState(), NOT_YET_AVAILABLE);
+        
+        List<EventStreamDay> week4Day4 = week4.getByDayEntries().get(3);
+        assertEquals(week4Day4.get(1).getTimeWindows().get(0).getState(), NOT_YET_AVAILABLE);
     }
     
     @Test
-    public void emptyEventStreamReportGeneratesEmptyStudyAdherenceReport() throws Exception {
-        AdherenceState state = new AdherenceState.Builder()
-                .withNow(DateTime.parse("2022-03-01T00:00:00.000-08:00")).build();
-        StudyAdherenceReport report = INSTANCE.generate(state);
+    public void generate_noEvents() throws Exception {
+        AdherenceState.Builder builder = createAdherenceState();
+        builder.withAdherenceRecords(createAdherenceRecords());
+        builder.withClientTimeZone("America/Chicago");
+        builder.withEvents(ImmutableList.of());
+        
+        StudyAdherenceReport report = INSTANCE.generate(builder.build());
         
-        // these fields are not set in the generator, they are set in the service.
-        // participant, testAccount, createdOn, timestamp, clientTimeZone
+        // testAccount is set in the service, not the generator
+        assertNull(report.getAdherencePercent());
+        assertEquals(report.getProgression(), ParticipantStudyProgress.UNSTARTED);
+        assertNull(report.getDateRange());
+        
+        assertEquals(report.getUnsetEventIds(), ImmutableSet.of("custom:event1", "timeline_retrieved", 
+                "study_burst:Study Burst:01", "study_burst:Study Burst:02", "study_burst:Study Burst:03"));
+        assertEquals(report.getUnscheduledSessions(), ImmutableSet.of("Supplemental Survey", "Initial Survey",
+                "Baseline Tapping Test", "Study Burst Tapping Test", "Final Survey"));
+        assertTrue(report.getEventTimestamps().isEmpty());
+
+        assertEquals(report.getWeeks().size(), 0);
+        
+        // Verify this too
+        assertNull(report.getWeekReport().getAdherencePercent());
```

Comment:
Is there more to verify about weekly report in this and other cases?

Reply:
No, this test is just about looking at the adherence percent and ensuring it is not generated where it should not be generated. This week is empty so there's no percent.